internal/batches/repozip/fetcher.go

```diff
@@ -13,63 +13,105 @@ import (
 
 	"github.com/cockroachdb/errors"
 
-	"github.com/sourcegraph/src-cli/internal/api"
-	"github.com/sourcegraph/src-cli/internal/batches/graphql"
+	"github.com/sourcegraph/src-cli/internal/batches/util"
 )
 
-// RepoFetcher abstracts the process of retrieving an archive for the given
+type RepoRevision struct {
+	RepoName string
+	Commit   string
+}
+
+// Fetcher abstracts the process of retrieving an archive for the given
 // repository.
-type RepoFetcher interface {
+type Fetcher interface {
 	// Checkout returns a RepoZip for the given repository and the given
 	// relative path in the repository. The RepoZip is possibly unfetched.
 	// Users need to call `Fetch()` on the RepoZip before using it and
 	// `Close()` once // they're done using it.
-	Checkout(repo *graphql.Repository, path string) RepoZip
+	Checkout(repo RepoRevision, path string) Archive
+}
+
+// Archive implementations represent a downloaded repository archive.
+type Archive interface {
+	// Fetch downloads the archive if it's not on disk yet.
+	Fetch(context.Context) error
```

Comment:
If we're renaming things we might want to rethink `Fetcher`, since the `Archive` itself has a `Fetch` method.

Ideas I just had while looking at this (meaning: they're not the best we could come up with, most likely): `Fetcher` could be `ArchiveRegistry` and `Archive.Fetch` could be `Ensure()` or something.

Reply:
Done 👍